wordpress-mobile/WordPress-Android
# Limit upload video duration on free plans

Please see this project pbArwn-3mr-p2 for context

<table><tr><th>Media</th><th>Editor</th></tr><tr><td><img src="https://user-images.githubusercontent.com/990349/146855087-0c19852c-f4d1-4c82-8401-2f8d2e9ab824.png" /></td><td><img src="https://user-images.githubusercontent.com/990349/146860677-656b9690-3d50-4c1f-8a40-3107dc2f5850.png" /></td></tr></table>

This PR aims to limit uploading videos longer than 5 minutes either taken using camera or while choosing files on device for the sites on free plan to prevent uploading larger files exploiting available 3GB overall limit.

To test:
- Select any site with a Free Plan
- Select Media from Quick links or Media under Publish on My Site
- Click on + button on top right hand corner in Media browser and select Choose from device
- Select a video file longer than 5 minutes in duration from media picker and click tick mark
- Notice that a toast message appears as shown in the first image above
- Try again Selecting a video file shorter than 5 min and notice it uploads without any message
- Also, try uploading a video taken using camera and the notice same rules apply there too
- Try also uploading a video through a Blog post or Story post with Gutenberg video block.  
- It should display message as in second image above for free sites if video is longer than 5 minutes.

## Regression Notes
1. Potential unintended areas of impact
Video uploads through media browser, video block from Gutenberg on blog and story posts

2. What I did to test those areas of impact (or what existing automated tests I relied on)
tested with various combinations on sites with free plan and paid plan

3. What automated tests I added (or what prevented me from doing so)
Updated existing unit tests

PR submission checklist:
